docs/design/w3c/spec.md

```diff
@@ -0,0 +1,545 @@
+presentation has a valid structure.
+
+In the case of W3C AnonCreds presentations, the `type` attribute includes an extra entry `AnonCredsPresentation`
+pointing to the difference in a base presentation structure and looks the following:
+
+```
+{
+  ... 
+  "type": [
+    "VerifiablePresentation",         // general verifiable presentation definition
+    "AnonCredsPresentation"           // definition for AnonCreds presentation
+  ]
+  ...
+}
+```
+
+#### Verifiable Credential
+
+W3C [Verifiable Credential](https://www.w3.org/TR/vc-data-model/#presentations-0)section requires including
+of `verifiableCredential` property to a verifiable presentation constructed from one or more verifiable credentials.
+
+The values of verifiable credentials are mostly constructed the same as described at
+the [Credential Structure](#credential) section.
+The only difference is the value of the `proof` property.
+
+In the case of W3C AnonCreds presentations, the `proof` attribute uses defined `AnonCredsPresentationProof2022`
+type pointing to the difference in a presentation structure and looks the following:
+
+```
+  "proof": {
+    "type": "AnonCredsPresentationProof2022",
+    "credential": {
+      "mapping": {
+        "revealedAttributes": [
+          {
+            "name": "firstName",
+            "referent": "attribute_0"
+          }
+        ],
+        "unrevealedAttributes": [
+          {
+            "name": "lastName",
+            "referent": "attribute_1"
+          }
+        ],
+        "requestedPredicates": [
+          {
+            "name": "age",
+            "p_type": "<",
+            "value": 18,
+            "referent": "predicate_1"
+          }
+        ]
+      },
+      "proofValue": "AAEBAnr2Ql...0UhJ-bIIdWFKVWxjU3ePxv_7HoY5pUw"
+    }
+  }
+```
+
+**Verifiable Credential Proof structure**
+
+* `proofValue` - encoded proof generated for each specific credential
+    * object created from Indy styled credential sub proof
+    ```
+        {
+            primaryProof: {..},
+            nonRevocProof: {..}
+        }
+    ```
+    * encoded
+      as [base64 attachment](https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0017-attachments#base64url).
+* `mapping` - date requested in the proof request
+    * data: attribute name and reference in the proof request
+        * `revealedAttributes` - list of requested attributes revealed using the credential
+        * `unrevealedAttributes` - list of requested attributes presented in the credential but left unrevealed
+        * `requestedPredicates` - list of predicates resolved using the credential
+
+
+* **TO DISCUSS**: Should we remove `mapping` completely or move under encoded `proofValue`?
+    * Why `mapping` is bad: we make presentation tied to Indy styled Presentation Request
+    * Mapping is something old indy-fashioned required for validation (not signature verification) that proof matches to
```

Comment:
The mapping is necessary for the verifier to align with the presentation request.  I don’t see how we can drop it.

Reply:
@swcurran  Please have a look at this pseudo code: https://github.com/hyperledger/anoncreds-rs/pull/266/files#diff-f0f0c92035decc44061ca415febaa763d3b4b86afc79572ba83f4d76d2a0f617R668

Reply:
I'm not sure what I'm looking at there -- the link takes me to all the files in the commit. How about we cover this on Monday -- you can show what you mean and we can discuss what happens if we don't include it.